lib/models/note_model.dart

```diff
@@ -1,20 +1,53 @@
-// ignore_for_file: public_member_api_docs, sort_constructors_first
 class Note {
-  int? id;
+  String id;
   String title;
   String content;
+  String updatedAt;
+  String? idUser;
+  bool isTrash;
 
-  Note({required this.id, required this.title, required this.content});
+  Note({required this.id, required this.title, required this.content, required this.updatedAt, this.idUser, required this.isTrash});
 
-  factory Note.fromMap(Map<String, dynamic> json) => Note(
-    id: json['id'],
+  factory Note.fromMapLocal(Map<String, dynamic> json) => Note(
+    id: json['id'].toString(),
     title: json['title'],
     content: json['content'],
+    updatedAt: json['updated_at'] ?? '',
+    idUser: json['id_user'] ?? '',
+    isTrash: json['is_trash'] == 1
   );
 
-  Map<String, dynamic> toMap() => {
+  factory Note.fromMapRemote(Map<String, dynamic> json) => Note(
```

Comment:
Although it's Map data type, but follow common standard, it's Json
So the name should be:
fromJson
fromRemoteJson
fromLocalJson
toLocalJson
...